pkg/tfgen/generate_nodejs.go

```diff
@@ -716,6 +715,9 @@ func (g *nodeJSGenerator) emitPlainOldType(w *tools.GenWriter, pot *propertyType
 		} else if prop.rawdoc != "" {
 			g.emitRawDocComment(w, prop.rawdoc, "    ")
 		}
+		if prop.deprecationMessage() != "" {
+			w.Writefmtln("    /** @deprecated %s */", prop.deprecationMessage())
```

Comment:
This is emitting a new doc comment for the property, which means any prior doc comment emitted by `g.emitDocComment` or `g.emitRawDocComment` won't be "associated" with the property as far as TypeDoc is concerned. This means the docs will show that the property is deprecated, but any other doc comments themselves will no longer show up in the docs for that property.

I could see how that might be OK -- after-all, the property is deprecated, so the doc comments are less relevant. But if we have docs for a property, I think it's still useful to show them, even if the property is deprecated.

Current (not ideal):

```typescript
/**
 * Important documentation.
 */
/** @deprecated use bar instead. */
readonly foo?: pulumi.Input<string>;
```

Proposed:

```typescript
/**
 * @deprecated use bar instead.
 *
 * Important documentation.
 */
readonly foo?: pulumi.Input<string>;
```

The fix would be to include the `@deprecated %s` as part of emitting the doc comment in `emitDocComment` and `emitRawDocComment`, possibly something like (passing the deprecation message to those functions):

```go
		if prop.doc != "" && prop.doc != elidedDocComment {
			g.emitDocComment(w, prop.doc, prop.docURL, prop.deprecationMessage(), "    ")
 		} else if prop.rawdoc != "" {
 			g.emitRawDocComment(w, prop.rawdoc, prop.deprecationMessage(), "    ")
 		} else if prop.deprecationMessage() != "" {
 			w.Writefmtln("    /** @deprecated %s */", prop.deprecationMessage())
 		}
```

And inside `emitDocComment` and `emitRawDocComment` if the passed-in deprecation message isn't `""`, emit it at the top with a blank comment line afterwards, so any further doc comments aren't included as part of the deprecation tag.

Reply:
@justinvp I just pushed the change you requested